xwiki-platform-core/xwiki-platform-refactoring/xwiki-platform-refactoring-api/src/main/java/org/xwiki/refactoring/link/LinkStore.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.refactoring.link;
+
+import java.util.Set;
+
+import org.xwiki.component.annotation.Role;
+import org.xwiki.model.reference.EntityReference;
+import org.xwiki.refactoring.RefactoringException;
+import org.xwiki.stability.Unstable;
+
+/**
+ * Allow accessing the links extracted from various entities.
+ * 
+ * @version $Id$
+ * @since 14.8RC1
+ */
+@Role
+@Unstable
+public interface LinkStore
+{
+    // TODO: also introduce an API (#getLinks) to get all types of links and not only entities
+
+    /**
+     * @param reference the reference of the entity containing links
+     * @return the other (DOCUMENT based) entities linked by the passed entity
+     * @throws RefactoringException when failing to load the links
+     */
+    Set<EntityReference> resolveLinkedEntities(EntityReference reference) throws RefactoringException;
```

Comment:
Looks like we were using previous "load" instead of "resolve" for same API method. Feels a bit weird to have "resolve" here, but wondering more why it's not "get"?

Reply:
It's not `get` because it does more than getting the stored links as is, it also makes sure to return DOCUMENT based references (so it has to resolve anything that is based on PAGE) so that you don't have to worry about that.